src/storage/mapping/ExtensionBasedMapper.ts

```diff
@@ -1,16 +1,24 @@
 import { promises as fsPromises } from 'fs';
 import * as mime from 'mime-types';
 import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
-import { TEXT_TURTLE } from '../../util/ContentTypes';
+import { APPLICATION_TRIG, TEXT_TURTLE } from '../../util/ContentTypes';
 import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
 import { joinFilePath, getExtension } from '../../util/PathUtil';
 import { BaseFileIdentifierMapper } from './BaseFileIdentifierMapper';
 import type { FileIdentifierMapperFactory, ResourceLink } from './FileIdentifierMapper';
 
 export class ExtensionBasedMapper extends BaseFileIdentifierMapper {
+  public static readonly mimeFallback: Record<string, string> = {
+    'application/trig': 'trig',
+  };
+
   private readonly types: Record<string, any>;
 
-  public constructor(base: string, rootFilepath: string, overrideTypes = { acl: TEXT_TURTLE, meta: TEXT_TURTLE }) {
+  public constructor(
+    base: string,
+    rootFilepath: string,
+    overrideTypes = { acl: TEXT_TURTLE, meta: TEXT_TURTLE, trig: APPLICATION_TRIG },
```

Comment:
```suggestion
    overrideTypes: Record<string, string> = { acl: TEXT_TURTLE, meta: TEXT_TURTLE, trig: APPLICATION_TRIG },
```
Not sure if this is needed but something that I wondered about for some time. I think otherwise Components.js might not allow non-default types here?

Reply:
It would only need to be updated in the files that are already being editted in this PR anyway, so I think that's fine for this PR.